### PR TITLE
De-nest docker registry auth and reformat related doc

### PR DESCRIPTION
**Pull request by carlosdp**
I'm not sure what happened here, but it looks like the Docker registry authentication parameters added in #390 aren't being decoded?

De-nesting the parameters seems to fix it. I'm guessing it's something to do with how `mapstructure` decodes the driver config struct.

I also removed the periods (`.`) from the related keys as they are awkward to type in HCL files as they have to be wrapped in quotes, and re-formatted the related docs a bit so that it's more clear the keys are at the same "level" as the rest of the driver config parameters. I can revert these though if there's a good reason they should be the way they were before.

### CODE REVIEW

**Comment by cbednarski** [comment]
Thanks for the PR. I'm not sure the implementation here was completed. I think it should look something like below and thus not need dots or underscores, but this mapstructure needs to be pinned to the `auth` key. Labels are also expressed this way now but since they're arbitrary k/v pairs there's no mapstructure associated with them.

```
config {
  auth {
    username = "username"
    password = "password"
  }
}
```

File: client/driver/docker.go [hunk 1]
```diff
@@ -392,6 +387,9 @@ func (d *DockerDriver) Start(ctx *ExecContext, task *structs.Task) (DriverHandle
 			ServerAddress: driverConfig.ServerAddress,
 		}
 
+		d.logger.Printf("[DEBUG] TASKCONFIG: %v", task.Config)
+		d.logger.Printf("[DEBUG] DRIVERCONFIG: %v", driverConfig)
+		d.logger.Printf("[DEBUG] AUTH: %v", authOptions)
```
Comment (cbednarski) [inline client/driver/docker.go]: This causes credential leakage.

**Comment by carlosdp** [comment]
I agree, that's how I originally thought it worked. I'll make that change.

File: client/driver/docker.go (hunk 1 above)
Comment (carlosdp) [inline client/driver/docker.go]: Yep, good catch

**Comment by carlosdp** [comment]
Updated to take keys under `auth` object.

**Comment by cbednarski** [comment]
Looks good! Thanks for following up. Would you be able to add a test for this?

**Comment by carlosdp** [comment]
Yep, was just thinking that this morning. Will add.

**Comment by cbednarski** [comment]
@carlosdp I'm going to merge this now because we want to avoid a BC break in config syntax in the 0.2 release. If you're still working on the tests please open a follow-up PR. Otherwise I will get around to this later. Thanks!

**Pull request merged**